.ci/Jenkinsfile

```diff
@@ -328,8 +337,10 @@ def generateFunctionalTestStep(Map args = [:]){
           }
           filebeat(output: "docker_logs_${suite}_${tags}.log", workdir: "${env.WORKSPACE}"){
             dir("${BASE_DIR}"){
-              withSecretVault(secret: "${VAULT_INSTRUMENTATION_SECRET}", user_key: "apmServerToken", user_var_name: "APM_SECRET_TOKEN", pass_key: "apmServerUrl", pass_var_name: "APM_SERVER_URL"){
-                sh script: """.ci/scripts/functional-test.sh "${suite}" "${tags}" "${STACK_VERSION}" "${BEAT_VERSION}" """, label: "Run functional tests for ${suite}:${tags}"
+              withEnv(["GOARCH=${goArch}"]) {
```

Comment:
@v1v do you think we can revert 0cd42e05badadaaa9e88c37ae4638033dcd3c8ff and apply that change to https://github.com/elastic/apm-pipeline-library/blob/6cfa9fac78ca050023ac0b5aa55de8d4090ff41c/vars/withGoEnvUnix.groovy#L48? Therefore, we will be setting the proper GOARCH value across any build.

Reply:
Will do this in a follow-up PR, thanks!